synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -2045,6 +2078,13 @@ def _find_previous_membership_txn(
                 Membership.BAN,
             )
 
+            if (
+                room_version_id is not None
+                and room_version_id not in KNOWN_ROOM_VERSIONS
+            ):
+                # Ignore rooms with unknown room versions (these were experimental rooms).
```

Comment:
Do we have some policy/documentation that explains how we should be handling rooms with unknown versions? Probably something that includes an explanation about how we can't deal with them because they might have different auth rules that we can't process properly without having support.

```suggestion
                # Ignore rooms with unknown room versions (these were experimental rooms that we no longer support).
```